src/pot.c

```diff
@@ -1,5 +1,22 @@
-#include <math.h>
 int pot(int a, int b){
-    a = pow(a, b);
-    return a;
+    int n = b, exp_bin = 0, resultado = 1;
+    if(b == 0){
+        return 1;
+    }
+    
+    for(int i=0, j=1;i<32;i++, j*=2){
+        if(b & (1 << i)){
+            while(exp_bin < j){
+                resultado = resultado * a;
+                exp_bin++;
+            }
+            exp_bin = 0;
```

Comment:
Perceba que se você mover esse `resultado = resultado * a` para fora desse loop, e no inicio do for, você não vai mais precisar desse loop e nem da variável `j`, o que vai deixar o código mais simples e legível.